uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -92,7 +94,9 @@ private static function validateAbsencePeriod($params) {
     //either there was an overlap or the absence period does not simply exist.
     if (!$period) {
       throw new InvalidLeaveRequestException(
-        "The Leave request dates is not contained within a valid absence period"
+        'The Leave request dates is not contained within a valid absence period',
+        'leave_request_not_within_absence_period',
+        'type_id'
```

Comment:
Instead of using type_id as the field, what about using from_date? I know sometimes the error will be because of the value in to_date, but at least this will be closer to what have caused the invalidation.